src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -315,70 +365,29 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
                 }
             })?;
 
-        if let Some((path, prim)) = is_primitive(&path_root, TypeNS) {
-            let impls =
-                primitive_impl(cx, &path).ok_or_else(|| ResolutionFailure::NotResolved {
+        let ty_res = if let Some(ty_res) = is_primitive(&path_root, TypeNS)
+            .map(|(_, res)| res)
+            .or_else(|| self.resolve_path(&path_root, TypeNS, module_id))
```

Comment:
After my spiel about separating refactors and changes, I don't want to make the change here, but I don't think these are both necessary:
```suggestion
        // FIXME: are these both necessary?
        let ty_res = if let Some(ty_res) = is_primitive(&path_root, TypeNS)
            .map(|(_, res)| res)
            .or_else(|| self.resolve_path(&path_root, TypeNS, module_id))
```

Reply:
In the strictest sense, they are, because removing is_primitive breaks tests, because modules aren't getting resolved. But even adding is_primitive back, it's only the modules that are called the same as the primitives that are... I wonder if this is because otherwise we'd end up returning early out of `resolve()`?

Reply:
Let's fix this in a follow-up.